akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -59,38 +59,82 @@ private[akka] final case class TakeWhile[T](p: T ⇒ Boolean, decider: Supervisi
 /**
  * INTERNAL API
  */
-private[akka] final case class DropWhile[T](p: T ⇒ Boolean, decider: Supervision.Decider) extends PushStage[T, T] {
+private[stream] final case class DropWhile[T](p: T ⇒ Boolean) extends GraphStage[FlowShape[T, T]] {
+  val in = Inlet[T]("DropWhile.in")
+  val out = Outlet[T]("DropWhile.out")
+  override val shape = FlowShape(in, out)
+  override def initialAttributes: Attributes = DefaultAttributes.dropWhile
   var taking = false
 
-  override def onPush(elem: T, ctx: Context[T]): SyncDirective =
-    if (taking || !p(elem)) {
-      taking = true
-      ctx.push(elem)
-    } else {
-      ctx.pull()
+  def createLogic(inheritedAttributes: Attributes) = new SupervisionedGraphStageLogic(inheritedAttributes, shape) with InHandler with OutHandler {
+    override def onPush(): Unit = execute(() ⇒ {
+      val el = grab(in)
+      if (taking || !p(el)) {
+        taking = true
+        push(out, el)
+      } else
+        pull(in)
+    })
+
+    override def onResume(t: Throwable): Unit = if (!hasBeenPulled(in)) pull(in)
+    override def onRestart(t: Throwable): Unit = onResume(t)
+
+    override def onPull(): Unit = pull(in)
+    setHandlers(in, out, this)
+  }
+  override def toString = "DropWhile"
+}
+
+/**
+  * INTERNAL API
+  */
+abstract private[stream] class SupervisionedGraphStageLogic(inheritedAttributes: Attributes, shape: Shape) extends GraphStageLogic(shape) {
+  val decider = inheritedAttributes.get[SupervisionStrategy].map(_.decider).getOrElse(Supervision.stoppingDecider)
+  def execute(f: () ⇒ Unit): Unit =
+    try { f() } catch {
+      case NonFatal(ex) ⇒ decider(ex) match {
+        case Supervision.Stop    ⇒ onStop(ex)
+        case Supervision.Resume  ⇒ onResume(ex)
+        case Supervision.Restart ⇒ onRestart(ex)
```

Comment:
The most common thing is `Stop`?I think how about：

1. Resume
2. Restart
3. Stop

？